src/modify.js

```diff
@@ -1,37 +1,85 @@
-// These are your event handlers
-const clickCounterHandler = () => {
+// Handles button clicks to increment the data-clicks attribute
+const clickCounterHandler = (e) => {
+  // Use event.target to access the clicked element directly
+  const button = e.target;
+
+  // Increment the data-clicks attribute (convert to a number first)
+  let clicks = Number(button.dataset.clicks) + 1;
+
+  // Update the data-clicks attribute
+  button.dataset.clicks = clicks;
+
+  // Update the button text to reflect the number of clicks
+  button.textContent =
+    clicks === 1
+      ? `I've been clicked ${clicks} time.`
+      : `I've been clicked ${clicks} times!`;
+
+  // Best Practice: Use template literals for string interpolation.
 };
 
-const handleKeydown = () => {
+// Updates the paragraph to display the last key pressed
+const handleKeydown = (e) => {
+  const keyDown = document.querySelector("#keydown-tracker");
+
+  // Display the key name using the event.code property
+  keyDown.textContent = `You pressed ${e.code}`;
+
 };
 
-// We've started this one for you
+// Handles event delegation for a container
 const handleDelegation = (event) => {
-  const resultSpan = document.querySelector('#delegation-result');
+  const resultSpan = document.querySelector("#delegation-result");
 
-  // TODO: Only do this if the target is right
-  resultSpan.textContent = event.target.textContent;
+  // Check if the target is a button
+  if (event.target.tagName === "BUTTON") {
+    resultSpan.textContent = event.target.textContent;
+  }
```

Comment:
You can also use `.matches` which I think is going to be more useful generally since it can take in any CSS selector

Also, I think this is a helpful moment to explain a bit more about WHY we need to check for this in the first place.

```suggestion
  // Since this handler will be called if we click anywhere within the `#delegation` container, we only want to update the text content if the target is a button
  if (event.target.matches('button') {
    resultSpan.textContent = event.target.textContent;
  }
```